packages/extension/src/hooks/use-extension/use-deeplink.js

```diff
@@ -3,17 +3,15 @@ import { useEffect } from 'react'
 
 const { queryParams } = Url
 
-export const useDeeplink = ({
-  appReady,
-  setPlayerIsOpen,
-  setSelectedSegment,
-}) => {
+export const useDeeplink = ({ appReady, setPlayerIsOpen }) => {
   useEffect(() => {
-    if (!appReady) {
-      const { play } = queryParams(window.location.href)
-      if (play) {
-        setPlayerIsOpen(true)
-      }
+    if (appReady) {
+      setTimeout(() => {
+        const { play } = queryParams(window.location.href)
+        if (play) {
+          setPlayerIsOpen(true)
+        }
+      }, 500)
```

Comment:
not a fan of this either, but it is the only mechanism I could get this working consistently. Tries `requestIdleCallback` but that only lead to some weird issues with the player starting at 4.